### PR TITLE
Align `metrics` summary method to `val` logs

### LINKED ISSUE #21170: Inconsistent validation metrics: Console output vs metrics.to_csv() results
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

Hello everyone,

I have written a small training script that performs a validation with the test dataset at the end and also saves the results in a CSV file. What I noticed is that the values in the CSV file do not match the values that are displayed in the console. It concerns the mAP values. Here is the script:

```Python
from ultralytics import YOLO
import os

# Load a model
model = YOLO("exported_last.pt")  # load a pretrained model (recommended for training)

# Paramter seettings
data="/raid/USERDATA/pawlodwp/dataset/labeled/Construction_Machines_v1/data.yaml"
batch=192 # 288 yolo11m
epochs=150 
imgsz=640 
device=[2,3]
cache=True
patience=20
project="yolo11l_machine_worker_pretrained"

# Train the model
results = model.train(data=data,
                      batch=batch, 
                      epochs=epochs, 
                      imgsz=imgsz, 
                      device=device, 
                      name="train",
                      project=project,
                      patience=patience,
                      cache=cache)
print("............")
metrics = model.val(data=data,
                    batch=batch,
                    imgsz=imgsz,
                    device=device,
                    name="test",
                    project=project, 
                    split="test")
print("............")

csv_str = metrics.to_csv()
csv_path = os.path.join(project, "validation_metrics.csv")

with open(csv_path, "w") as f:
    f.write(csv_str)
```

This is the output in the console:

Image

This is the CSV:

[validation_metrics.csv](https://github.com/user-attachments/files/20878082/validation_metrics.csv)

I would like to have the same values in the same output in the CSV file. Including the average values across all classes (all) as in the console output. Does anyone have an idea?

### Additional

_No response_

### PR DESCRIPTION
**Pull request by RizwanMunawar**
## 🛠️ PR Summary

<sub>Made with ❤️ by [Ultralytics Actions](https://github.com/ultralytics/actions)<sub>

### 🌟 Summary
Improved validation result summaries and documentation for easier analysis and export of YOLO model metrics. 📊✨

### 📊 Key Changes
- Added a direct Colab link in the docs to demonstrate exporting validation results to formats like DataFrame, CSV, and SQL.
- Updated the `summary()` output in metrics utilities:
  - Renamed metric keys for clarity and consistency (e.g., `Box-P`, `Mask-P`, `Pose-P`).
  - Included more detailed per-class information, such as number of images and instances.
  - Standardized classification summary keys to `top1_acc` and `top5_acc`.
  - Removed some redundant or less-used scalar metrics from the summary output.

### 🎯 Purpose & Impact
- 📈 Makes validation results easier to interpret, compare, and export for further analysis.
- 🧑‍💻 Helps both developers and non-experts quickly access and understand key performance metrics.
- 🚀 Enhances documentation with practical resources, streamlining workflows for users who want to analyze or share results in various formats.

### CODE REVIEW

**Comment by UltralyticsAssistant** [comment]
👋 Hello @RizwanMunawar, thank you for submitting an `ultralytics/ultralytics` 🚀 PR! To ensure a seamless integration of your work, please review the following checklist:

- ✅ **Define a Purpose**: Clearly explain the purpose of your fix or feature in your PR description, and link to any [relevant issues](https://github.com/ultralytics/ultralytics/issues). Ensure your commit messages are clear, concise, and adhere to the project's conventions.
- ✅ **Synchronize with Source**: Confirm your PR is synchronized with the `ultralytics/ultralytics` `main` branch. If it's behind, update it by clicking the 'Update branch' button or by running `git pull` and `git merge main` locally.
- ✅ **Ensure CI Checks Pass**: Verify all Ultralytics [Continuous Integration (CI)](https://docs.ultralytics.com/help/CI/) checks are passing. If any checks fail, please address the issues.
- ✅ **Update Documentation**: Update the relevant [documentation](https://docs.ultralytics.com/) for any new or modified features.
- ✅ **Add Tests**: If applicable, include or update tests to cover your changes, and confirm that all tests are passing.
- ✅ **Sign the CLA**: Please ensure you have signed our [Contributor License Agreement](https://docs.ultralytics.com/help/CLA/) if this is your first Ultralytics PR by writing "I have read the CLA Document and I sign the CLA" in a new message.
- ✅ **Minimize Changes**: Limit your changes to the **minimum** necessary for your bug fix or feature addition. _"It is not daily increase but daily decrease, hack away the unessential. The closer to the source, the less wastage there is."_ — Bruce Lee

For more guidance, please refer to our [Contributing Guide](https://docs.ultralytics.com/help/contributing/). Don’t hesitate to leave a comment if you have any questions. Thank you for contributing to Ultralytics! 🚀

_This is an automated response; an Ultralytics engineer will assist you soon._

**Comment by RizwanMunawar** [comment]
@Laughing-q Hi, this PR will align the validation data (after converting it into different formats, i.e, Pandas `DataFrame`, CSV, HTML) with the `val` logs we display directly on the console. Currently, it's missing images and instances information used during validation, also adjusted the per-class AP (Before it was mAP). Please note, changes of this PR are only in the `metrics.py`. Also, users request this feature as well. #21170.

#### Example-01: Display validation logs

**Code**

```python
from ultralytics import YOLO

model = YOLO("yolo11n.pt")

if __name__ == '__main__':
    results = model.val(data="coco8.yaml", batch=4, workers=1)
```

**Results**

```bash
Ultralytics 8.3.159  Python-3.11.0 torch-2.5.0+cpu CPU (Intel Core(TM) i5-10400 2.90GHz)
YOLO11n summary (fused): 100 layers, 2,616,248 parameters, 0 gradients, 6.5 GFLOPs
val: Fast image access  (ping: 0.00.0 ms, read: 640.7334.8 MB/s, size: 54.0 KB)
val: Scanning G:\Ultralytics Work\datasets\coco8\labels\val.cache... 4 images, 0 backgrounds, 0 corrupt: 100%|██████████| 4/4 [00:00<?, ?it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 1/1 [00:00<00:00,  3.38it/s]
                   all          4         17       0.57       0.85      0.847      0.632
                person          3         10      0.557        0.6      0.585      0.272
                   dog          1          1      0.548          1      0.995      0.697
                 horse          1          2      0.531          1      0.995      0.674
              elephant          1          2      0.371        0.5      0.516      0.256
              umbrella          1          1      0.569          1      0.995      0.995
          potted plant          1          1      0.847          1      0.995      0.895
Speed: 1.7ms preprocess, 63.0ms inference, 0.0ms loss, 1.5ms postprocess per image
Results saved to G:\Ultralytics Work\ultralytics\runs\detect\val
```

#### Example-02: Display the `metrics.to_df()` logs

*Code*

```python
from ultralytics import YOLO

model = YOLO("yolo11n.pt")

if __name__ == '__main__':
    results = model.val(data="coco8.yaml", batch=4, workers=1)
    print(results.to_df())
```

**Results**

```bash
          Class  Images  Instances    Box-P  Box-R   Box-F1    mAP50  mAP50-95
0        person       3         10  0.55724    0.6  0.57783  0.58518   0.27239
1           dog       1          1  0.54802    1.0  0.70803  0.99500   0.69650
2         horse       1          2  0.53058    1.0  0.69331  0.99500   0.67398
3      elephant       1          2  0.37078    0.5  0.42580  0.51583   0.25634
4      umbrella       1          1  0.56900    1.0  0.72531  0.99500   0.99500
5  potted plant       1          1  0.84717    1.0  0.91726  0.99500   0.89550
```

Please let me know what you think. Thanks

Edit: turns out, we can not add spaces in the variable names, i.e,` Box (P),` due to `to_xml()` restrictions, so I adjusted each space with `-`

**Comment by Laughing-q** [comment]
@RizwanMunawar got it! looks good. Can you also update the notebook accordingly?
pic-250626-1549-32
And let me know once you updated, then we could go ahead and merge this one

**Comment by RizwanMunawar** [comment]
@Laughing-q, ok, got it. Actually, it won't be possible as the notebook uses the Ultralytics latest version, and this change is currently not in our latest version. Once the PR is merged and a new version is released, I will immediately update the notebook. However, I just performed testing in the notebook by using this branch, and everything is working fine.

```bash
!git clone https://github.com/ultralytics/ultralytics/
%cd "ultralytics"
!git checkout metrics
!pip install -e . 
```

Screenshot 2025-06-26 130529

**Review by Laughing-q — APPROVED** [review]
ok sounds good

**Comment by UltralyticsAssistant** [comment]
🎉 Fantastic work, @RizwanMunawar and @Laughing-q! Your improvements to validation result summaries and documentation are a huge step forward for the Ultralytics community. As Leonardo da Vinci said, "Simplicity is the ultimate sophistication." By making YOLO metric analysis clearer and more accessible, you've truly empowered users at every level. Thank you for your thoughtful contributions—your efforts make Ultralytics HUB and YOLO12 even more impactful for everyone! 🚀
